arduino/WandCore/WandCore.ino

```diff
@@ -253,72 +257,114 @@ void loop()
     incrementRGBColorloop();
   }
 
+  // Read IMU sensor
+  readIMU(imuData);
+
   // Read the touch surface and handle the event
+  // Take immediate action if single tap or double tap; if holding, action happens only if gesture is detected
   int currentTouchEvent = getTouchEvent();
-  switch (currentTouchEvent)
-  {
-    case SINGLE_TAP:
-      // trigger color sensor
-      debugI("single tap happened");
-      
-      break;
-    case DOUBLE_TAP:
-      debugI("double tap happened");
-      // change mode
-      if (wandMode == POWER) {
-        wandMode = COLOR;
-        incrementRGBColorloop(); 
-        debugI("Switched to Color Mode");
-      } else {
-        wandMode = POWER;
-        setRGBLED(255, 255, 255); // white for power mode
-        debugI("Switched to Power Mode"); 
-      }      
-      
-      break;
-    case HOLDING:
-      // TODO adjust brightness or color depending on mode
-      if ((millis() - lastLogged) >= 500) { // DEBUG ONLY, throttling log output
-        debugI("touch is being held");
-        lastLogged = millis();
-      }
+  if (currentTouchEvent == SINGLE_TAP) {
+    debugI("Single Tap Just Happened");
+    if (wandMode == POWER) {                      // power mode, single tap == set light brightness based on "brightness" of color sensed

```

Comment:
> // power mode, single tap == set light brightness based on "brightness" of color sensed

I see you also put in the Miro "define what this means", and I'm not sure there is a good definition? I'm not sure we need to have a single tap function in this state.

Reply:
We could do nothing, though I think we could also use Saturation or Value as a proxy for brightness relatively easily? i just hadn't picked something yet. would need to do an RGB to HSV conversion.

Reply:
We don't get anything other than hue information from the color sensor -- saturation and brightness/value are independent of hue.

Reply:
hmm. doesn't the "clear" value from the sensor influence RGB in some way? or otherwise give us a dimension beside hue?

Reply:
Oh, I see, that was really buried in the docs/API. I suppose it does give a value of intensity, but I'm still not sure it'd be that worthwhile. Practical use of this seems iffy, what's a good usecase?

Reply:
lol it's a toy wand, not sure "practical" is a primary lens here; that said, it's not the most compelling, so we can just scratch it. the use case would be touching a darker color (e.g. black, dark blue) and setting a dim brightness, or a lighter color (white, light pink, etc) and setting a higher brightness.

Reply:
I meant practical in the sense that we probably don't want the wand to have behaviors that are unclear, especially because it is a toy and a 3yo is the target audience! 

I haven't seen the clear sensor in action, but from reading I don't think that's how it works. It's a separate sensor from RGB, so a light source (regardless of black / white color) would read a higher value than not a light source (even if it was white, light pink, etc.). And we don't have good brightness control of the wand RGB LED so we couldn't display feedback very well.